File: firka/ios/HomeWidgetsExtension/Models/Subject.swift
import Foundation

struct WidgetSubject: Codable {
    let uid: String
    let name: String
    let category: NameUidDesc?
    let sortIndex: Int
    let teacherName: String?
}

struct NameUidDesc: Codable {
    let uid: String
    let name: String
    let description: String?
}

File: kreta_api/lib/src/model/subject.dart
import 'generic.dart';

class Subject extends NameUid {
  final NameUidDesc category;
  final int sortIndex;

  Subject({
    required super.uid,
    required super.name,
    required this.category,
    required this.sortIndex,
  });

  factory Subject.fromJson(Map<String, dynamic> json) {
    return Subject(
      uid: json['Uid'],
      name: json['Nev'],
      category: json.nameUidDesc('Kategoria')!,
      sortIndex: json['SortIndex'],
    );
  }

  @override
  String toString() {
    return 'Subject('
        'uid: "$uid", '
        'name: "$name", '
        'category: $category, '
        'sortIndex: $sortIndex, '
        ')';
  }
}
